vendor/bat-native-ads/src/bat/ads/internal/ads/ad_notifications/ad_notification_exclusion_rules.cc

```diff
@@ -36,103 +35,118 @@ ExclusionRules::ExclusionRules(
     resource::AntiTargeting* anti_targeting_resource,
     const AdEventList& ad_events,
     const BrowsingHistoryList& browsing_history)
-    : subdivision_targeting_(subdivision_targeting),
-      anti_targeting_resource_(anti_targeting_resource),
-      ad_events_(ad_events),
-      browsing_history_(browsing_history) {
-  DCHECK(subdivision_targeting_);
-  DCHECK(anti_targeting_resource_);
-}
+    : ExclusionRulesBase(subdivision_targeting,
+                         anti_targeting_resource,
+                         ad_events,
+                         browsing_history) {}
 
 ExclusionRules::~ExclusionRules() = default;
 
 bool ExclusionRules::ShouldExcludeCreativeAd(
-    const CreativeAdInfo& creative_ad) const {
-  bool should_exclude = false;
+    const CreativeAdInfo& creative_ad) {
+  if (WasExcluded(creative_ad)) {
+    return true;
+  }
 
   DailyCapFrequencyCap daily_cap_frequency_cap(ad_events_);
   if (ShouldExclude(creative_ad, &daily_cap_frequency_cap)) {
-    should_exclude = true;
+    Exclude(daily_cap_frequency_cap.GetUuid(creative_ad));
```

Comment:
seems like `ShouldExclude` is not a proper name of this function now.
maybe we can rename function to somthing like `ExcludeIfNeeded` ?